doc/whats_new/v0.23.rst

```diff
@@ -197,6 +197,11 @@ Changelog
 
 - |Enhancement| improve error message in :func:`utils.validation.column_or_1d`.
   :pr:`15926` by :user:`Loïc Estève <lesteve>`.
+
+- |Enhancement| `optimize._newton_cg` will now check for inferior or equal
```

Comment:
Should we regard this as a bug fix? I'd rather this say "will avoid an unnecessary iteration by checking for ..." as this is a more relevant framing for users.

Reply:
(Is there a way to construct a case where we do indeed redduce the number of iterations?)

Reply:
>(Is there a way to construct a case where we do indeed reduce the number of iterations?)

well, I am not sure if there is...

Reply:
I have changed the whatsnew